.github/workflows/pre-merge-integration-test.yml

```diff
@@ -52,10 +50,10 @@ jobs:
         run: sam deploy --config-file infrastructure/samconfig.toml --stack-name ${{ env.STACK_NAME }}
 
       - name: Run mocked tests
```

Comment:
Since you're in this space...

Can you move the mocked tests to run before the `sam deploy`? Since mocked tests don't run in AWS, it won't create a stack and fail to delete if the mocked tests fail.

Reply:
Great optimisation - I've applied it to the PR 👍